rte/main.go

```diff
@@ -225,20 +229,41 @@ func defaultHostName() string {
 	return val
 }
 
-func defaultTopologyManagerPolicy() string {
-	if val, ok := os.LookupEnv("TOPOLOGY_MANAGER_POLICY"); ok {
-		return val
+func setupTopologyManagerConfig(pArgs *ProgArgs, conf config.Config) error {
+	policyFrom := "args"
+	// do not overwrite with empty an existing value (e.g. from opts)
+	if pArgs.RTE.TopologyManagerPolicy == "" {
+		policyFrom = "env"
+		pArgs.RTE.TopologyManagerPolicy = os.Getenv("TOPOLOGY_MANAGER_POLICY")
+	}
+	// still no luck? last try from conf
+	if pArgs.RTE.TopologyManagerPolicy == "" {
+		policyFrom = "conf"
+		pArgs.RTE.TopologyManagerPolicy = conf.TopologyManagerPolicy
+	}
+	if pArgs.RTE.TopologyManagerPolicy == "" {
+		return fmt.Errorf("cannot find setting for Topology Manager Policy")
 	}
-	// empty string is a valid value here, so just keep going
-	return ""
-}
 
-func defaultTopologyManagerScope() string {
-	if val, ok := os.LookupEnv("TOPOLOGY_MANAGER_SCOPE"); ok {
-		return val
+	scopeFrom := "args"
+	if pArgs.RTE.TopologyManagerScope == "" {
+		scopeFrom = "env"
+		pArgs.RTE.TopologyManagerScope = os.Getenv("TOPOLOGY_MANAGER_SCOPE")
 	}
-	//https://kubernetes.io/docs/tasks/administer-cluster/topology-manager/#topology-manager-scopes
-	return "container"
+	// still no luck? last try from conf
+	if pArgs.RTE.TopologyManagerScope == "" {
+		scopeFrom = "conf"
+		pArgs.RTE.TopologyManagerScope = conf.TopologyManagerScope
+	}
+	if pArgs.RTE.TopologyManagerScope == "" {
+		return fmt.Errorf("cannot find setting for Topology Manager Scope")
```

Comment:
why not return the default?

Reply:
same for the tmpolicy check above

Reply:
good point. I'm still a bit torn between fail loudly and being able to fallback to sane defaults. I'll put some more thought in this.